frontend/src/concepts/pipelines/content/compareRuns/CompareRunsContext.tsx

```diff
@@ -38,32 +38,38 @@ export const CompareRunsContextProvider = conditionalArea<CompareRunsContextProv
     const runIds = searchParams.get(CompareRunsSearchParam.RUNS);
     return runIds ? runIds.split(',') : [];
   }, [searchParams]);
+  const loadedRef = React.useRef(false);
```

Comment:
why switch to using a ref here and not the loaded state from useFetchState? It looks like now on initial navigation to the page, the error/empty state is flashed instead of the loading indicator

Reply:
i am guessing this is becuase loaded is set before runs is set, causing a few frames to be incorrect

Reply:
Once the compare runs page loads, that useFetchState's "loaded" state never changes from "true" even when the array of IDs changes. If you want to solve this some other way, let me know.

Reply:
thats correct, it wont. thats just how useFetchState works. loaded will be true when the data is not an error and has loaded before. You should be able to react on runs changing instead of having the loaded ref flip.

Reply:
I don't understand what you're suggesting, but your if-statement in the effect added here; https://github.com/opendatahub-io/odh-dashboard/blob/b7576b6a87b63daf8479d5a498595c7b77938092/frontend/src/concepts/pipelines/content/compareRuns/CompareRunsContext.tsx#L64 is checking for a difference in runs already. This effect triggers before the request for new runs occurs when switching from Manage runs to Compare runs. 

This means that we get invalid runs and those run IDs in the URL get stripped before the request even gets made. How would you like to solve this problem?

Reply:
Alternatively, if we don't have Manage runs be a sub-route of Compare runs, the whole context would re-render when visiting Compare runs (coming from Manage runs) and this wouldn't be an issue.

Reply:
lets do that. move the context under each page respectively. really the context will be mostly useful in the first compare runs page as it will have many components relying on the selections